### DISCUSSION TITLE
nvim-chezmoi - A neovim plugin that offers chezmoi integration

### DISCUSSION THREAD

**Discussion by andre-kotake** [twpayne/chezmoi#4049] [Show and tell]
Hi!

I recently wrote a NeoVim plugin entirely in Lua that provides chezmoi integration for nvim, you can check it out here:

https://github.com/andre-kotake/nvim-chezmoi

My main motivation for this was getting annoyed everytime I opened a source file with nvim due to losing all the good stuff like LSP, completions and syntax highlights because of the wrong filetype being applied to the file.

This plugins tries to find the correct filetype for a source file using the target file name whenever you try to edit a source file. It does so by calling chezmoi commands in order to find the target file name.

I also provide a small API for executing and getting the results for chezmoi commands and also some ready to use nvim commands. Chezmoi command results are cached to speed up future invocations of the same command in a session, but you may also force the execution always.

Some of the stuff this plugins offers:
- `:ChezmoiEdit` for opening source files anytime from anywhere.
- `:ChezmoiExecuteTemplate` for previewing your file template in a new buffer.
- `:ChezmoiManaged` to list all your managed files (telescope extension).

Full documentation in [README](https://github.com/andre-kotake/nvim-chezmoi/blob/main/README.md)

Feedback would be greatly appreciated!

**Comment by twpayne** [comment]
Very nice, thank you very much!

I've added a link to nvim-chezmoi to chezmoi's website in ba12f36f20b0ce4c80a68b4e16f5e1ac9d6ede53.
